OpenRA.Mods.Common/Warheads/ChangeOwnerWarhead.cs

```diff
@@ -27,6 +27,9 @@ public class ChangeOwnerWarhead : Warhead
 		public override void DoImpact(Target target, WarheadArgs args)
 		{
 			var firedBy = args.SourceActor;
+			if (!target.IsValidFor(firedBy))
```

Comment:
Shouldn't this be checked inside the loop over the actors? Just because the direct target is invalid doesn't mean that all the actors in the impact zone are invalid.

Reply:
Yes, you are right. Compare `GrantExternalConditionWarhead`.